### ISSUE TITLE
Status

### ISSUE THREAD

**Issue by bhack** [wbenbihi/hourglasstensorlfow#2]
Hi I'am interested to experiment this TF version. What is the status of your implementation? 
What accurancy can you already reproduce?
Do you plan to train/test over [Posetrack](https://posetrack.net)?

**Comment by bhack** [comment]
Other extensions over hourglass are available in https://github.com/bearpaw/torch-pose

**Comment by wbenbihi** [comment]
The implementation is done for the Hourglass Network, I should combine it with an object detection model (tiny YOLO is the network I am focusing on right now) to proceed to multiple person pose estimation. I tried it with a pretrained (on 20 classes) YOLO model but the bounding boxes are coarse. I am working on this.
Regarding the accuracy, I should use the PCK benchmark, unfortunately I have some issue with my Matlab version. So, it should be done in few days. 
Accuracy is currently measured during validation. I can share the TensorBoard event log if you want.
For every image a HeatMap is outputed (shape=[numberOfStacks, 64, 64, numberOfJoints]). To get values between 0 and 1 I use a sigmoid and then for every joint, the argmax is computed.
I measured accuracy as follow : joint_accur_p = 1 - sqrt((xp_pred - xp_gt)^2 +  (yp_pred - yp_gt)^2 )/sqrt(2*64^2)

It is a basic L2 (Euclidean) distance. I reached an average of 90% with this . 
But to be clear, it was just an indicator to myself, I don't trust that measurement to be a valid proof of the model efficiency.

**Comment by bhack** [comment]
Yes probably a better evaluation is required. Probably could be interesting to add a summary of the (activations) in tensorboard so it could be possibile to visual inspect activations on training.
For the detector there is also a faster-rcnn model available on the shelf in TF. Probably it is slower than Yolo but you can give a try.

**Comment by wbenbihi** [comment]
I am curious about the activations summary. Could you elaborate? Should I add last tensor of every stack in the summary to visualize this? Or Add every layer of the last stack to the summary? I never tried to add tensors to summary so my questions might be weird! 
Concerning the faster-rcnn, I investigate a lot on it (also on Mask-RCNN), I had some troubles with the anchors. I found some model already implemented in TF but they were too slow. With the tiny YOLO I reach 10-12 FPS on a nVidia Geforce 1070.

**Comment by bhack** [comment]
[Here](https://github.com/CharlesShang/FastMaskRCNN/issues/99) and in the code you can find an example of bounding box summary on the tensorboard. But you can draw whatever you want to visual monitor the output at training time i.e. different keypoint activation on train/val.

**Comment by bhack** [comment]
With top down approaches, so people bboxes, probably you will not recover what you lost from the detector performance. So or you use a very performing detector or you need to try with bottom up approaches.

**Comment by bhack** [comment]
You can see the Top-down Bottom-up approach in https://arxiv.org/pdf/1612.01465.pdf
